### PR TITLE
Make test target name unique.

### PR DESCRIPTION
**Pull request by gberardi-pillar**
everest-core cannot build with tests (BUILD_TESTING=ON) due to submodules having generic names that collide when CMake runs.

This change makes the test target for libevse-security unique by naming it after the repo, a pattern that can be followed for all repos.

NOTE: This change does not enable automated builds and tests, as there are other PRs to handle that work.
I will also point out that when I run the tests manually, 17 out of 26 tests fail.

ALSO NOTE: I changed the flag from BUILD_TESTING_EVSE_SECURITY to BUILD_TESTING, since that variable gets used by CTest and is needed to allow the make test target to work.

I also used the Modern CMake pattern of enabling tests only if the project is the main project (not a child of a parent project) or if it is explicitly specified with `cmake -Deverest-evse_security_BUILD_TESTING=ON` to allow flexibility.

### CODE REVIEW

**Comment by gberardi-pillar** [comment]
@shankari Here's a minor change to review.

**Comment by gberardi-pillar** [comment]
@valentin-dimov Since you have contributed the most to this repo, would you be able to look at this PR?

File: CMakeLists.txt
Comment (Dominik-K) [inline CMakeLists.txt:56]:
```suggestion
if( (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING) OR ${PROJECT_NAME}_BUILD_TESTING)
```

**Review by Dominik-K — CHANGES_REQUESTED** [review]

**Comment by Dominik-K** [comment]
Let's discuss in general at

- https://github.com/EVerest/EVerest/issues/129

**Review by shankari — APPROVED** [review]
I don't understand the reason for the comment
https://github.com/EVerest/libevse-security/pull/28#discussion_r1452279593
but I have asked that as a question in
https://github.com/EVerest/EVerest/issues/129#issuecomment-1897848657

I don't see any other questions here, and it would be good to make more progress on cleaning up the tests.

I approve assuming that you will address the comment assuming that it is warranted by the outcome of the larger discussion before merge.

File: CMakeLists.txt
Comment (gberardi-pillar) [reply CMakeLists.txt:56]: I am following the pattern in the Modern CMake book: https://cliutils.gitlab.io/modern-cmake/chapters/testing.html

Comment (andistorm) [inline CMakeLists.txt:56]:
```suggestion
if((${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME} OR ${PROJECT_NAME}_BUILD_TESTING) AND BUILD_TESTING)
```

File: CMakeLists.txt
Comment (andistorm) [inline CMakeLists.txt:1]: Please add options
```

option(${PROJECT_NAME}_BUILD_TESTING "Build unit tests, used if included as dependency" ON)
option(BUILD_TESTING "Build unit tests, used if standalone project" ON)
```

**Review by andistorm — CHANGES_REQUESTED** [review]

**Review by andistorm — APPROVED** [review]

**Pull request merged**
